src/history/hs_rec.c

```diff
@@ -213,10 +219,24 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
         error_on_ts_ordering = false;
     }
 
-    if (ret == 0)
-        WT_ERR(__hs_delete_reinsert_from_pos(session, cursor, btree->id, key, tw->start_ts + 1,
-          true, false, error_on_ts_ordering, &counter, tw));
-
+    if (ret == 0) {
+        WT_ASSERT(session, tw->start_ts + 1 > WT_TS_NONE);
+        /* Get the history store cursor position. */
+        WT_ERR_NOTFOUND_OK(__hs_cursor_pos(session, cursor, btree->id, key, tw->start_ts + 1,
+                             error_on_ts_ordering, tw, &non_ts_updates),
+          true);
+
+        if (ret == 0) {
+            /* For non-timestamped updates we only need the counter, skip reinsert. */
+            if (non_ts_updates)
+                __hs_insert_pos_counter(session, cursor, btree->id, key, &counter);
```

Comment:
The counter is already adjusted at line 141. We don't need another adjustment here. Let me know If I missed any scenario where this code is required.

Reply:
Hi Hari, good spotting - we did not see the counter was already set. We saw that in `hs_delete_reinsert_from_pos` was setting the counter (because it keeps track when reinserting), but now that we skip this whole function for `non_ts_updates`, if the counter is already present then we don't need it.

Reply:
Removed here [b6900cf](https://github.com/wiredtiger/wiredtiger/pull/10686/commits/b6900cf433be480879bbfe2be8d51f4507fcea13)